flake.nix

```
@@ -108,7 +108,7 @@
         }
         ''
           mkdir -p "$out/bin"
-          $CXX -Wall -pedantic ${demos.chat-client.cpp.src} \
+          $CXX -Wall -I"./demos/chat-client/c++/assets.h" -pedantic ${demos.chat-client.cpp.src} \
```

```suggestion
          $CXX -Wall -I"${demos.chat-client.cpp.src}/assets.h" -pedantic ${demos.chat-client.cpp.src}/fibonacci.cpp \
```

Will allow `nix` to access `assets.h` if you set `demos.fibonacci.cpp.src = ./demos/fibonacci/c++;` on L37